branch_name_verification.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+import robot.basestation.app as app
+from robot.basestation.app import run_shell
+import re
+
+excluded_branches = {'master', 'develop', 'staging', 'test'}
+
+output, error = run_shell("git symbolic-ref --short HEAD", "", False)
+branch_name = output.decode("utf8")[0:-1]
+
+def is_excluded_branch(branch_name, excluded_branches):
+    """
+    check if branch name is in list of excluded branches
+    """
+    return branch_name in excluded_branches
+
+def has_upper_case_letters(trim_issue_num):
+    """
+    check if trimed branch name only contains lower case characters
+    """
+    return not (trim_issue_num.islower())
+
+def has_hyphen_seperator(trim_issue_num):
+    """
+    check for hyphen seperator between branch name and issue number
+    """
+    return trim_issue_num.endswith('-')
+
+def has_issue_num(branch_name):
+    """
+    check if there is an issue number
+    """
+    return not re.search(r'\d+$', branch_name) is None
```

Comment:
I recommend making this function retrieve the issue number and return it, if it is not there return None.

This will allow you to make unit tests to verify that it actually returns the correct number which at the moment you do not test.

If you really wanna keep this function then you can just do 

```
def has_issue_num(branch_name):
  return get_issue_num(branch_name) != None